src/rayleigh_quotient.jl

```diff
@@ -85,12 +88,35 @@ end
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_EIG)
     b = prob.b
     C = prob.C
-    Kb = I - b * b' / dot(b, b)
-    nlargest = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
     N = size(b, 1)
-    inds = map(!in(nlargest), 1:N)
-    J = I(N)[inds, :]
+    Kb = I - b * b' / dot(b, b)
+    _, inds_keep = _inds_to_keep_and_remove(b, N)
+    J = I(N)[inds_keep, :]
     augC = J * Kb * C
+    return _solve_homo_prob_and_normalize(augC, prob, alg)
+end
+
+function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_SPARSE)
+    b = prob.b
+    C = prob.C
+    N = size(b, 1)
+    inds_remove, inds_keep = _inds_to_keep_and_remove(b, N)
+    J = I(N)[inds_keep, :]
```

Comment:
```suggestion
    J = sparse(I(N))[inds_keep, :]
```
We should make sure everything is sparse

Reply:
Could we just slice the matrix directly instead of multiplying with J?

Reply:
yes, that makes sense I think